api/src/pcapi/routes/backoffice/user_deletion/blueprint.py

```diff
@@ -0,0 +1,56 @@
+import logging
+
+from flask import flash
+from flask import redirect
+from flask import url_for
+from markupsafe import Markup
+from sqlalchemy.exc import IntegrityError
+from werkzeug.exceptions import NotFound
+
+from pcapi import settings
+from pcapi.core.permissions import models as perm_models
+from pcapi.core.users import models as users_models
+from pcapi.models import db
+from pcapi.repository import atomic
+from pcapi.repository import mark_transaction_as_invalid
+from pcapi.routes.backoffice import utils
+from pcapi.routes.backoffice.users.blueprint import users_blueprint
+
+
+logger = logging.getLogger(__name__)
+
+
+@users_blueprint.route("/<int:user_id>/delete", methods=["POST"])
+@utils.permission_required_in([perm_models.Permissions.SUSPEND_USER])
+@atomic()
+def delete_user(user_id: int) -> utils.BackofficeResponse:
+    if not settings.ENABLE_USER_DELETION:
+        raise NotFound()
+
+    user = (
+        users_models.User.query.filter_by(id=user_id).populate_existing().with_for_update(key_share=True).one_or_none()
```

Comment:
[question] Pourquoi `key_share=True` ? Si je comprends bien, ça n'empêche pas un `update`, ou même un autre `select for key share`. De quoi cherche-t-on à se prémunir, ici ? 

(*Disclaimer :* Je ne suis pas spécialiste des locks sur `select`.)